Ginger/GingerCore/ALM/OctaneCore.cs

```diff
@@ -0,0 +1,159 @@
+﻿using ALM_Common.DataContracts;
+using Amdocs.Ginger.Common;
+using Amdocs.Ginger.Repository;
+using GingerCore.Activities;
+using GingerCore.Variables;
+using OctaneSDK.Connector;
+using OctaneSDK.Connector.Authentication;
+using OctaneSDK.Connector.Credentials;
+using OctaneSDK.Services;
+using OctaneSDK.Services.RequestContext;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Configuration;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GingerCore.ALM
+{
+    public class OctaneCore : ALMCore
+    {
+        public override ObservableList<ActivitiesGroup> GingerActivitiesGroupsRepo { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
+        public override ObservableList<Activity> GingerActivitiesRepo { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
+
+        public RestConnector mOctaneRestConnector;
+        public EntityService entityService;
+        protected LwssoAuthenticationStrategy lwssoAuthenticationStrategy;
+        protected WorkspaceContext workspaceContext;
+        protected SharedSpaceContext sharedSpaceContext;
+
+        public override bool ConnectALMProject()
+        {
+            throw new NotImplementedException();
+        }
+
+        public override bool ConnectALMServer()
+        {
+            try
+            {
+                string host;
+                string userName;
+                string password;
+                Reporter.ToLog(eLogLevel.DEBUG, "Connecting to qTest server");
+                mOctaneRestConnector = new RestConnector();
+                entityService = new EntityService(mOctaneRestConnector);
+                if (!mOctaneRestConnector.IsConnected())
+                {
+                    string ignoreServerCertificateValidation = "false"; //ConfigurationManager.AppSettings["ignoreServerCertificateValidation"];
+                    if (ignoreServerCertificateValidation != null && ignoreServerCertificateValidation.ToLower().Equals("true"))
+                    {
+                        NetworkSettings.IgnoreServerCertificateValidation();
+                    }
+                    NetworkSettings.EnableAllSecurityProtocols();
+
+                    host = ALMCore.DefaultAlmConfig.ALMServerURL;
+
+                    // If webAppUrl is empty we do not try to connect.
+                    if (string.IsNullOrWhiteSpace(host)) return false;
```

Comment:
Codacy found an issue: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6200394)